src/extractor/extractor_callbacks.cpp

```diff
@@ -175,6 +175,38 @@ void ExtractorCallbacks::ProcessWay(const osmium::Way &input_way, const Extracti
         }
     }
 
+    const auto classStringToMask = [this](const std::string &class_name) {
+        auto iter = classes_map.find(class_name);
+        if (iter == classes_map.end())
```

Comment:
Can we use the string map for deduplicating both strings as well as the classes here? Or better: I don't understand why we need another deduplication map.